File: src/main/java/com/patrickanker/chatterbox/core/messengers/PlayerMessenger.java
package com.patrickanker.chatterbox.core.messengers;

import com.patrickanker.chatterbox.api.Messenger;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import org.bukkit.entity.Player;

import java.util.Optional;
import java.util.UUID;

public class PlayerMessenger implements Messenger {

    private final UUID uuid;

    public PlayerMessenger(UUID id) {
        uuid = id;
    }

    /**
     * The ID of the PlayerMessenger is the UUID of the associated player. The value of the UUID is stored locally so that Chatterbox can check to see if the player is still online.
     *
     * @return The UUID of the player in String form
     */
    @Override
    public String getID() {
        return uuid.toString();
    }

    @Override
    public String getFocus() {
        return null;
    }

    @Override
    public void assignFocus(String channelID) {

    }

    public Optional<OfflinePlayer> getOfflinePlayer() {
        return Optional.ofNullable(Bukkit.getOfflinePlayer(uuid));
    }

    public Optional<Player> getPlayer() {
        return Optional.ofNullable(Bukkit.getOfflinePlayer(uuid).getPlayer());
    }
}

File: src/main/kotlin/chatterbox/core/messengers/PlayerMessenger.kt
package chatterbox.core.messengers

import chatterbox.api.Messenger
import org.bukkit.Bukkit
import org.bukkit.OfflinePlayer
import org.bukkit.entity.Player
import java.util.*

class PlayerMessenger constructor(private val uuid: UUID) : Messenger {
    override var focus: String
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
        set(value) {}

    override fun assignFocus(channelID: String) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override val id: String
        get() = uuid.toString()

    var offlinePlayer: OfflinePlayer? = null
        get() = Bukkit.getOfflinePlayer(uuid)

    var onlinePlayer: Player? = null
        get() = offlinePlayer?.player
}